### PR TITLE
Add opencensus native stuff

### PR DESCRIPTION
**Pull request by ibawt**
# What?
- Gem structure for opentelemetry - mimicing 'concurrent-ruby', ie base gem + -ruby-ext for c extensions
- ext folder with build working for opencensus proto grpc definitions

### CODE REVIEW

File: opentelemetry-ruby-ext.gemspec [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'opentelemetry/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = 'opentelemetry-ruby-ext'
+  spec.version       = OpenTelemetry::VERSION
+  spec.authors       = ['Ian Quick']
+  spec.email         = ['ian.quick@shopify.com']
+
+  spec.summary       = 'C extensions for opentelemetry client library'
```
Comment (fbogsany) [inline opentelemetry-ruby-ext.gemspec]: Technically incorrect 😄

File: opentelemetry-ruby-ext.gemspec [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+  spec.homepage      = 'https://opentelemetry.com'
```
Comment (fbogsany) [inline opentelemetry-ruby-ext.gemspec]: Should be `.io`

File: opentelemetry-ruby-ext.gemspec [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  spec.license       = 'Apache'
+
+  spec.files         = Dir['ext/**/*.{h,c,cc}']
+  spec.require_paths = ['lib']
+  spec.extensions    = 'ext/opentelemetry-ruby-ext/extconf.rb'
+
+  spec.required_ruby_version = '>= 2.3.0'
```
Comment (fbogsany) [inline opentelemetry-ruby-ext.gemspec]: I'd like to target `>= 2.5.0`

File: opentelemetry-ruby.gemspec [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# frozen_string_literal: true
+
+lib = File.expand_path('lib', __dir__)
+$LOAD_PATH.unshift(lib) unless $LOAD_PATH.include?(lib)
+require 'opentelemetry/version'
+
+Gem::Specification.new do |spec|
+  spec.name          = 'opentelemetry-ruby'
+  spec.version       = OpenTelemetry::VERSION
+  spec.authors       = ['Ian Quick', 'Francis Bogsyani']
```
Comment (fbogsany) [inline opentelemetry-ruby.gemspec]: s/Bogsyani/Bogsanyi/

File: opentelemetry-ruby.gemspec [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  spec.email         = ['ian.quick@shopify.com']
+
+  spec.summary       = 'OpenTelemetry client library'
+  spec.homepage      = 'https://opentelemetry.com'
```
Comment (fbogsany) [inline opentelemetry-ruby.gemspec]: `.io`

File: opentelemetry-ruby.gemspec [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  spec.license       = 'Apache'
+
+  spec.files         = Dir.chdir(File.expand_path('.', __dir__)) do
+    `git ls-files -z`.split("\x0").reject do |f|
+      f.match(%r{^(test|spec|features)/})
+    end
+  end
+  spec.require_paths = ['lib']
+  spec.required_ruby_version = '> 2.3.0'
```
Comment (fbogsany) [inline opentelemetry-ruby.gemspec]: `>= 2.5.0`

File: opentelemetry-ruby-ext.gemspec (hunk 1 above)
Comment (ibawt) [reply opentelemetry-ruby-ext.gemspec]: lulz, I'll change it to 'native'

File: opentelemetry-ruby.gemspec (hunk 4 above)
Comment (ibawt) [reply opentelemetry-ruby.gemspec]: I totally didn't look up your name and did it from memory.  :(

File: opentelemetry-ruby.gemspec (hunk 6 above)
Comment (ibawt) [reply opentelemetry-ruby.gemspec]: won't be able use mutexes then lolololol

**Comment by ibawt** [comment]
I don't like this approach, going to can it.

**Pull request closed without merging**